src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/services/service-form/service-form.component.html

```diff
@@ -685,6 +685,61 @@
             </div>
           </div>
         </ng-container>
+        <!-- Grafana -->
+        <ng-container *ngIf="serviceForm.controls.service_type.value === 'grafana'">
+          <div class="form-group row">
+            <label class="cd-col-form-label"
+                   for="grafana_port">
+              <span i18n>Grafana Port</span>
+              <cd-helper>
+                <span i18n>The default port used by grafana.</span>
+              </cd-helper>
+            </label>
+            <div class="cd-col-form-input">
+              <input id="grafana_port"
+                     class="form-control"
+                     type="number"
+                     formControlName="grafana_port"
+                     min="1"
+                     max="65535">
+              <span class="invalid-feedback"
+                    *ngIf="serviceForm.showError('grafana_port', frm, 'pattern')"
+                    i18n>The entered value needs to be a number.</span>
+              <span class="invalid-feedback"
+                    *ngIf="serviceForm.showError('grafana_port', frm, 'min')"
+                    i18n>The value must be at least 1.</span>
+              <span class="invalid-feedback"
+                    *ngIf="serviceForm.showError('grafana_port', frm, 'max')"
+                    i18n>The value cannot exceed 65535.</span>
+              <span class="invalid-feedback"
+                    *ngIf="serviceForm.showError('grafana_port', frm, 'required')"
+                    i18n>This field is required.</span>
+            </div>
+          </div>
+
+          <div class="form-group row">
+            <label i18n
+                   class="cd-col-form-label"
+                   for="grafana_admin_password">Initial admin password</label>
+            <div class="cd-col-form-input">
+              <div class="input-group">
+                <input id="grafana_admin_password"
+                       class="form-control"
+                       type="password"
+                       autocomplete="new-password"
+                       formControlName="grafana_admin_password">
+                <span class="input-group-append">
+                  <button type="button"
+                          class="btn btn-light"
+                          cdPasswordButton="grafana_admin_password">
+                  </button>
+                  <cd-copy-2-clipboard-button source="grafana_admin_password">
+                  </cd-copy-2-clipboard-button>
+                </span>
+              </div>
+            </div>
+          </div>
+        </ng-container>
```

Comment:
Rather than calling it "initial admin password" (I understand that's the Cephadm wording), I'd just call it "admin password" (that's the [Grafana setting](https://grafana.com/docs/grafana/latest/setup-grafana/configure-grafana/#admin_password)) and instead I'd ensure that it can only be changed on service creation, not on subsequent edits (by disabling the input field). 

If a clarification is needed we could add an info popover ℹ️ with some explanation:
```
The password of the default Grafana Admin. Set once on first-run. Default is admin.
```

Reply:
Thanks. I changed the name to `Grafana Password`

> I'd ensure that it can only be changed on service creation, not on subsequent edits (by disabling the input field).

why though? this can still be done from cephadm by updating the spec so disabling from the dashboard doesn't make much sense right? and even if the field is disabled, if someone wants to they can still edit the service by deleting and creating it again..

Reply:
Thanks @nizamial09 !

Assuming the password is [only set when Grafana runs for the first time](https://grafana.com/docs/grafana/latest/setup-grafana/configure-grafana/#admin_password), this could be quite misleading. In fact if you modify it and, for example, you grow the Grafana service placement from 1 to 3 nodes, the 1st node will have the original password, while the other 2 nodes will end up having a different one.

Changing the admin password requires calling [`grafana-cli admin reset-admin-password`](https://grafana.com/docs/grafana/latest/cli/#reset-admin-password), but I don't see that in Cephadm code. What's your view on this, @adk3798 @rkachach ?

Reply:
@epuertat I went through some old PRs and discussions happened in it. And it makes sense now to disable the password field in the edit form. So I made the changes to disable it. Thanks

Reply:
Thanks!